AdminUi/apps/admin_ui/lib/home/clients_overview/clients_filter.dart

```diff
@@ -0,0 +1,188 @@
+import 'package:admin_api_sdk/admin_api_sdk.dart';
+import 'package:admin_api_types/admin_api_types.dart';
+import 'package:admin_ui/core/core.dart';
+import 'package:flutter/material.dart';
+import 'package:get_it/get_it.dart';
+import 'package:multi_dropdown/multiselect_dropdown.dart';
+
+class ClientsFilter {
+  final String? clientId;
+  final String? displayName;
+  final List<String>? tiers;
+  final (FilterOperator, DateTime)? createdAt;
+  final (FilterOperator, int)? numberOfIdentities;
+
+  const ClientsFilter({
+    this.clientId,
+    this.displayName,
+    this.tiers,
+    this.createdAt,
+    this.numberOfIdentities,
+  });
+
+  static const empty = ClientsFilter();
+
+  ClientsFilter copyWith({
```

Comment:
It's just filtering a list of X with some filter params. If you really want to get into it we can go over it.

Reply:
The current UI looks as follows:
![image](https://github.com/nmshd/backbone/assets/65653426/813a0a06-5dac-48e1-adeb-412e92f2a40f)

As you can see, each column has a filter. Depending on the type of the data in the column, the filter looks different. 
* In case of a simple string (Client ID, Display Name), it is just a text box, and we perform a plain text search on the property mapped to the column. 
* In case of a finite list of possibilities (Default Tier), we offer a dropdown list. 
* In case of a number (Number of Identities), there are two fields: a dropdown with a comparison operator (=, <, >, <=, >=) and a text field where you can enter a number
* In case of a date (Created At), there are two fields: a dropdown with a comparison operator (=, <, >, <=, >=) and a date picker.